packages/@uppy/tus/src/index.js

```diff
@@ -278,10 +277,16 @@ export default class Tus extends BasePlugin {
         resolve(upload)
       }
 
-      uploadOptions.onShouldRetry = (err) => {
+      if (this.opts.onShouldRetry !== null) {
+        uploadOptions.onShouldRetry = (err) => opts.onShouldRetry(err, defaultOnShouldRetry)
+      } else {
+        uploadOptions.onShouldRetry = defaultOnShouldRetry
+      }
+
+      function defaultOnShouldRetry (err) {
```

Comment:
Can we avoid using the function before it's declared?

Reply:
I think hoisting isn't bad practice when it comes to functions. It shows you the usage before the implementation details.